packages/rollup-dev-tools/test/transpiler/abi-encoded-constants-transpilation.spec.ts

```diff
@@ -0,0 +1,132 @@
+  getPUSHIntegerOp,
+} from '../../src'
+import {
+  ErroredTranspilation,
+  OpcodeReplacer,
+  OpcodeWhitelist,
+  TranspilationErrors,
+  TranspilationResult,
+  Transpiler,
+  SuccessfulTranspilation,
+} from '../../src/types/transpiler'
+import {
+  TranspilerImpl,
+  OpcodeReplacerImpl,
+  OpcodeWhitelistImpl,
+} from '../../src/tools/transpiler'
+import { transpileAndDeployInitcode, mockSSTOREReplacer } from '../helpers'
+
+const log = getLogger(`test-constructor-params-new`)
+const abi = new ethers.utils.AbiCoder()
+
+const getGetterReturnedVal = async (
+  deployedAddress: Buffer,
+  methodId: string,
+  evmUtil: EvmIntrospectionUtil
+): Promise<Buffer> => {
+  const callRes: ExecutionResult = await evmUtil.callContract(
+    bufToHexString(deployedAddress),
+    methodId
+  )
+  if (!!callRes.error) {
+    throw new Error(
+      `call to ${methodId} failed with evmUtil Error: ${callRes.error}`
+    )
+  }
+  return callRes.result
+}
+
+describe('Solitity contracts should have hardcoded values correctly accessible in transpiled initcode', () => {
+  let evmUtil: EvmIntrospectionUtil
+
+  const opcodeWhitelist = new OpcodeWhitelistImpl(Opcode.ALL_OP_CODES)
+  const transpiler = new TranspilerImpl(opcodeWhitelist, mockSSTOREReplacer)
+  let deployedGetterAddress: Buffer
+  beforeEach(async () => {
+    evmUtil = await EvmIntrospectionUtilImpl.create()
+    log.debug(
+      `transpiling and deploying initcode which should store hash in constructor`
+    )
+    deployedGetterAddress = await transpileAndDeployInitcode(
+      AbiEncodedConstantInConstructor,
+      [],
+      [],
+      transpiler,
+      evmUtil
+    )
+    const code: Buffer = await evmUtil.getContractDeployedBytecode(
+      deployedGetterAddress
+    )
+    log.debug(
+      `Initcode transpiled and deployed.  The code is:\n${formatBytecode(
+        bufferToBytecode(code)
+      )}`
+    )
+  })
+
+  it(`The hash of an abi.encode(hardcoded) should be correct and retrievable if stoed during constructor()`, async () => {
+    // set up expected values step-by-step
+    const expectedInnerBytesBeingHashed = Buffer.from(
+      ethers.utils.toUtf8Bytes(
+        'EIP712Domain(string name,string version,uint256 chainId,address verifyingContract)'
+      )
+    )
+    log.debug(
+      `expectedInnerBytesBeingHashed: ${bufToHexString(
+        expectedInnerBytesBeingHashed
+      )}`
+    )
+    const expectedInnerHashRaw = ethers.utils.keccak256(
+      expectedInnerBytesBeingHashed
+    )
+    log.debug(`expectedInnerHashRaw: ${expectedInnerHashRaw}`)
+    const expectedInnerHashAndValEncoded = ethers.utils.defaultAbiCoder.encode(
+      ['bytes32', 'uint256'],
+      [expectedInnerHashRaw, 1]
+    )
+    log.debug(
+      `expectedInnerHashAndValEncoded: ${expectedInnerHashAndValEncoded}`
+    )
+    const expectedOuterHash = ethers.utils.keccak256(
+      expectedInnerHashAndValEncoded
+    )
+    log.debug(`expected final outer hash: ${expectedOuterHash}`)
```

Comment:
might be cleaner to just do this all in one go and not log the intermediate values (e.g. https://github.com/ethereum-optimism/uniswap-v2-core/blob/706c0ef8749511d9c36a05e459a2ace9099c85ce/test/UniswapV2ERC20.spec.ts#L37-L52)

Reply:
Ah yeah, this was set up for debugging to figure out which stage it was failing at.  Removing